### DISCUSSION TITLE
Running Transmission in headless daemon

### DISCUSSION THREAD

**Discussion by falconmick** [transmission/transmission#5283] [Q&A]
I'm looking to run Transmission as a BitTorrent client for a native app I'm building for windows/macOS.

I saw that the RPC stuff which looks very promising and I was wondering if there is some way to execute transmission as a headless daemon so that if the user does or does not have transmission on their system my copy won't interest with their copy and won't require them to download a copy.

Cheers 😁 

When I get the chance I'll have a play either way but I was hoping I could be led down a good path given the docs are in progress

**Comment by ckerr** [comment]
Unless I'm misunderstanding the question, `transmission-daemon` sounds like what you're looking for?

**Reply by falconmick** [reply]
Hey I've been making lots of very tiered posts so maybe that's why this isn't very sensible haha, yeah that 90% what I'm asking and exactly what I need, next part is if it's possible to run that daemon without installing though the msi or dmg files but just to get a portable version of some assortment. My guess is either I'll just build transmission from source and use the transmission daemon that way OR I might be forced to look further into how those portable versions of transmission that are out there make the app able to run without instal.

This is important because the usecase is me bundling a non installed executable background process of the transmission daemon without users having to install or remove transmission from their systems, basically they should have no idea it's using transmission unless they look into licensing or check out the processes tab.

I'll try to dig further and leave my findings tomorrow

**Reply by bluskye** [reply]
Sounds like you just want it in docker and have users access daemon via webpage ?  The stock UI is pretty simple though vs the ones from 3.0.  Transmission in docker is how I use transmission 100% from my file server.

https://fleet.linuxserver.io/image?name=linuxserver/transmission
